autogen/agentchat/conversable_agent.py

```diff
@@ -768,9 +877,6 @@ def generate_reply(
             logger.error(error_msg)
             raise AssertionError(error_msg)
 
-        if messages is None:
```

Comment:
Note: I removed the two lines here (need to confirm):
```
        if messages is None:
            messages = self._oai_messages[sender]
```
Why we can delete this: 
The two lines deleted is in every `generate_<>_reply` function. So when both messages and sender are passed to a subsequent `generate_<>_reply`, it will perform the same logic.
```
final, reply = reply_func(self, messages=messages, sender=sender, config=reply_func_tuple["config"])
``` 

Why needed for compression: Compression will modify `self._oai_messages`, and it is expected that `generate_oai_reply` will use the updated messages from `self._oai_messages`. With the two lines, the messages will not be None and the updated `self._oai_messages` will not be used.

Reply:
Both justifications are not clear to me.

Reply:
Another option is to only remove this when `compress_config` is provided. This ensures the original logic is unchanged in the default seting